examples/ui/node_transform.rs

```diff
@@ -0,0 +1,388 @@
+    mut translation_query: Query<&mut NodeTranslation, With<TargetNode>>,
+) {
+    let controls = [
+        (KeyCode::Left, -Vec2::X),
+        (KeyCode::Right, Vec2::X),
+        (KeyCode::Up, -Vec2::Y),
+        (KeyCode::Down, Vec2::Y),
+    ];
+    for &(key_code, direction) in &controls {
+        if input.pressed(key_code) {
+            for mut translation in translation_query.iter_mut() {
+                translation.0 += direction * 50.0 * time.delta_seconds();
+                translation.0 = translation.0.clamp(Vec2::splat(-150.), Vec2::splat(150.));
+            }
+        }
+    }
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    // ui camera
+    commands.spawn(Camera2dBundle::default());
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                flex_basis: Val::Percent(100.0),
+                align_items: AlignItems::Center,
+                justify_content: JustifyContent::Center,
+                ..default()
+            },
+            background_color: Color::BLACK.into(),
+            ..default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn(NodeBundle {
+                    style: Style {
+                        align_items: AlignItems::Center,
+                        justify_content: JustifyContent::SpaceEvenly,
+                        gap: Size::all(Val::Px(25.)),
+                        ..default()
+                    },
+                    ..default()
+                })
+                .with_children(|parent| {
+                    parent
+                        .spawn(NodeBundle {
+                            style: Style {
+                                flex_direction: FlexDirection::Column,
+                                justify_content: JustifyContent::Center,
+                                padding: UiRect::all(Val::Px(10.)),
+                                gap: Size::all(Val::Px(10.)),
+                                ..default()
+                            },
+                            background_color: Color::BLACK.into(),
+                            z_index: ZIndex::Global(1),
+                            ..default()
+                        })
+                        .with_children(|parent| {
+                            parent
+                                .spawn(ButtonBundle {
+                                    style: Style {
+                                        size: Size::all(Val::Px(50.)),
+                                        align_items: AlignItems::Center,
+                                        justify_content: JustifyContent::Center,
+                                        ..default()
+                                    },
+                                    background_color: Color::WHITE.into(),
+                                    ..default()
+                                })
+                                .insert(RotateButton(-std::f32::consts::PI / 8.))
```

Comment:
```suggestion
                                .insert(RotateButton(-std::f32::consts::FRAC_PI_8))
```
You should use the appropriate constants instead of dividing PI
see https://doc.rust-lang.org/std/f32/consts/index.html

Reply:
please also do the other PI fractions